### ISSUE TITLE
linux openEuler

### ISSUE THREAD

**Issue by wfc66986** [tauri-apps/tauri#7787]
### Describe the bug

warning: `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "libsoup-2.4" "libsoup-2.4 >= 2.62"` did not exit successfully: exit status: 1

error: failed to run custom build command for `soup2-sys v0.2.0`

Caused by:
  process didn't exit successfully: `/root/Desktop/xx/src-tauri/target/debug/build/soup2-sys-5b0e13e12053653d/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=LIBSOUP_2.4_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "libsoup-2.4" "libsoup-2.4 >= 2.62"` did not exit successfully: exit status: 1
  error: could not find system library 'libsoup-2.4' required by the 'soup2-sys' crate

  --- stderr
  Package libsoup-2.4 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libsoup-2.4.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libsoup-2.4', required by 'virtual:world', not found
  Package 'libsoup-2.4', required by 'virtual:world', not found

warning: build failed, waiting for other jobs to finish...

### Platform and versions

```text
warning: `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "libsoup-2.4" "libsoup-2.4 >= 2.62"` did not exit successfully: exit status: 1

error: failed to run custom build command for `soup2-sys v0.2.0`

Caused by:
  process didn't exit successfully: `/root/Desktop/xx/src-tauri/target/debug/build/soup2-sys-5b0e13e12053653d/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=LIBSOUP_2.4_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" "pkg-config" "--libs" "--cflags" "libsoup-2.4" "libsoup-2.4 >= 2.62"` did not exit successfully: exit status: 1
  error: could not find system library 'libsoup-2.4' required by the 'soup2-sys' crate

  --- stderr
  Package libsoup-2.4 was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libsoup-2.4.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libsoup-2.4', required by 'virtual:world', not found
  Package 'libsoup-2.4', required by 'virtual:world', not found

warning: build failed, waiting for other jobs to finish...
```

**Comment by FabianLars** [comment]
Did you install these dependencies? https://tauri.app/v1/guides/getting-started/prerequisites#setting-up-linux - i don't know what their names are on openeuler but they are still required.

If it still doesn't work maybe this is the issue? https://tauri.app/v1/guides/faq#build-conflict-with-homebrew-on-linux

**Comment by wfc66986** [comment]
我安装了，但是装不上去在openEluer欧拉上。说是没有这个包 我看官方提供的只能在ubuntu上安装成功运行，有什么办法解决吗。



---Original---
From: ***@***.***&gt;
Date: Fri, Sep 8, 2023 22:10 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [tauri-apps/tauri] linux openEuler (Issue #7787)




 
Did you install these dependencies? https://tauri.app/v1/guides/getting-started/prerequisites#setting-up-linux - i don't know what their names are on openeuler but they are still required.
 
If it still doesn't work maybe this is the issue? https://tauri.app/v1/guides/faq#build-conflict-with-homebrew-on-linux

**Comment by FabianLars** [comment]
So i just spun up a VM running openeuler and the RHEL instructions from our website linked above worked just fine.
So in short, this should be enough:
```sh
sudo dnf install webkit2gtk3-devel
sudo dnf groub install "Development Tools"
```

**Comment by wfc66986** [comment]
好的 明天我试下想问下可以加个微信联系方式嘛方便更好沟通。



---Original---
From: ***@***.***&gt;
Date: Sat, Sep 9, 2023 00:20 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [tauri-apps/tauri] linux openEuler (Issue #7787)




 
So i just spun up a VM running openeuler and the RHEL instructions from our website linked above worked just fine.
 So in short, this should be enough:
 sudo dnf install webkit2gtk3-devel sudo dnf groub install "Development Tools"

**Comment by wfc66986** [comment]
安装你的方法&nbsp;sudo dnf install webkit2gtk3-devel这个是可以运行的，但是打包后生成deb文件运行跑不起来，麻烦帮忙看下谢谢。








fc
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tauri-apps/tauri"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月9日(星期六) 凌晨0:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tauri-apps/tauri] linux openEuler (Issue #7787)





 
So i just spun up a VM running openeuler and the RHEL instructions from our website linked above worked just fine.
 So in short, this should be enough:
 sudo dnf install webkit2gtk3-devel sudo dnf groub install "Development Tools"

**Comment by wfc66986** [comment]
运行pnpm tauri dev 报错了
在openEluer linux上


fc
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tauri-apps/tauri"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月9日(星期六) 凌晨0:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tauri-apps/tauri] linux openEuler (Issue #7787)





 
So i just spun up a VM running openeuler and the RHEL instructions from our website linked above worked just fine.
 So in short, this should be enough:
 sudo dnf install webkit2gtk3-devel sudo dnf groub install "Development Tools"

**Comment by wfc66986** [comment]
如何解决

**Comment by scambier** [comment]
Hello, visitors from the future.

- I'm using Linux Mint (fresh install)
- pnpm
- VSCode
- All prerequisites installed
- `pnpm tauri dev` fails, but `pnpm tauri build` works

What worked for me:

- Don't use the embedded terminal in VSCode. I guess there are some messed up env vars
- Re-run `pnpm tauri dev` from a "real" terminal to get another error message that says ``linking with `cc` failed: exit status: 1``
- `sudo apt remove build-essential && sudo apt install build-essential`
- Reboot
- Aaaand it works.
